lib/foreman_inventory_upload/generators/fact_helpers.rb

```diff
@@ -61,7 +61,7 @@ def obfuscate_hostname?(host)
         insights_client_setting = ActiveModel::Type::Boolean.new.cast(insights_client_setting)
         return insights_client_setting unless insights_client_setting.nil?
 
-        Setting[:obfuscate_inventory_hostnames]
+        CommonParameter.find_by(name: 'obfuscate_inventory_hostnames')&.value
```

Comment:
You should look for a value on the host, not just the common parameter, similar to https://github.com/theforeman/foreman_rh_cloud/blob/48f9d74ba8a671c74cec3a79b2646c38172a42a3/app/controllers/insights_cloud/api/machine_telemetries_controller.rb#L87